### PR TITLE
Bb 347 base class

### PR DESCRIPTION
**Pull request by bastbijl**
@jonathansmit7 @gvdent @malamili

### CODE REVIEW

File: nifi-weaver-processors/src/test/java/com/weaverplatform/nifi/XmiImporterTest.java [hunk 1]
```diff
@@ -32,8 +33,6 @@
 import java.io.InputStream;
 import java.util.List;
 
-import static org.junit.Assert.*;
-
```
Comment (gvdent) [inline nifi-weaver-processors/src/test/java/com/weaverplatform/nifi/XmiImporterTest.java]: A Test should really have asserts, the fact that this import can be removed is troubling :/

File: nifi-weaver-processors/src/main/java/com/weaverplatform/nifi/individual/XmiImporter.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+
+    session.read(flowFile, new InputStreamCallback() {
+
+      @Override
+      public void process(InputStream inputStream) throws IOException {
+
+        // TODO: call xmi importer
+
+      }
```
Comment (gvdent) [inline nifi-weaver-processors/src/main/java/com/weaverplatform/nifi/individual/XmiImporter.java]: Less whitespace please. If a method relies on using double enters to make it more structured, that is a hint that the method needs to be split up into smaller parts.

File: nifi-weaver-processors/src/main/java/com/weaverplatform/nifi/individual/CreateValueProperty.java [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+    descriptors.add(OBJECT_ATTRIBUTE);
+    descriptors.add(OBJECT_STATIC);
+    this.properties = Collections.unmodifiableList(descriptors);
+    
+    relationshipSet.add(ORIGINAL);
+    this.relationships = new AtomicReference<>(relationshipSet);
+  }
+
+  @Override
+  public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
```
Comment (gvdent) [inline nifi-weaver-processors/src/main/java/com/weaverplatform/nifi/individual/CreateValueProperty.java]: This function is too long, extract functionality to its own methods. There are also 3 blocks of code which look similar enough to smell like a DRY violation.

File: nifi-weaver-processors/src/main/java/com/weaverplatform/nifi/individual/CreateIndividualProperty.java [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+    descriptors.add(OBJECT_ATTRIBUTE);
+    descriptors.add(OBJECT_STATIC);
+    this.properties = Collections.unmodifiableList(descriptors);
+
+    relationshipSet.add(ORIGINAL);
+    this.relationships = new AtomicReference<>(relationshipSet);
+  }
+
+  @Override
+  public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
```
Comment (gvdent) [inline nifi-weaver-processors/src/main/java/com/weaverplatform/nifi/individual/CreateIndividualProperty.java]: onTrigger method is too long, should have blocks of functionality extracted to its own methods.

File: nifi-weaver-processors/src/main/java/com/weaverplatform/nifi/individual/CreateIndividualProperty.java [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+    String subjectAttributeValue = context.getProperty(SUBJECT_ATTRIBUTE).getValue();
+    String subjectStaticValue    = context.getProperty(SUBJECT_STATIC).getValue();
+    if(subjectAttributeValue != null) {
+      subject = flowFile.getAttribute(subjectAttributeValue);
+    } else if(subjectStaticValue != null) {
+      subject = subjectStaticValue;
+    }
+    if(subject == null) {
+      throw new ProcessException("No subject found for this Individual Property.");
+    }
```
Comment (gvdent) [inline nifi-weaver-processors/src/main/java/com/weaverplatform/nifi/individual/CreateIndividualProperty.java]: The previous block of code looks similar to the next two blocks, and are straight up copy/paste from elsewhere. Extract similar code to a shared superclass or other utility methods for ease of maintenance.

**Comment by gvdent** [comment]
This pull request contains a deletion of an existing test class, and no addition of new test classes. This should not block the pull request, but does indicate the project needs some testing TLC, especially since NIFI components ought to be easily isolated and tested.

**Comment by jonathansmit7** [comment]
top!

File: nifi-weaver-processors/src/test/java/com/weaverplatform/nifi/XmiImporterTest.java (hunk 1 above)
Comment (bastbijl) [inline nifi-weaver-processors/src/test/java/com/weaverplatform/nifi/XmiImporterTest.java]: agreed, the xmi-importer will be added later, and tested later too

File: nifi-weaver-processors/src/main/java/com/weaverplatform/nifi/individual/XmiImporter.java (hunk 2 above)
Comment (bastbijl) [inline nifi-weaver-processors/src/main/java/com/weaverplatform/nifi/individual/XmiImporter.java]: Fixed

**Comment by gvdent** [comment]
Nice cleanup! Merge on!

**Pull request merged**